### PR TITLE
Evaluate `imports` to discover which modules should be recursively included in Nix-Gui

### PR DESCRIPTION
**Pull request by lapp0**
- Evaluate `imports` to resolve import errors resulting from context-requiring expressions which `OptionDefinition` cannot handle (such as `modulesPath + "/installer/scan/not-detected.nix"`) (see `parser.py`)
- Pass the correct `modulesPath` in `evalModuleStub` (see `lib.nix`)
- Refactor `nix_eval` so there is a generic way to call `nix-instantiate` (`nix_eval.py`)
- Implement `nix_eval.resolve_nix_search_path` so `<nixpkgs/nixos/modules>` can be resolved. (`nix_eval.py`)
- Update `option_definition.Path` to use `nix_eval.resolve_nix_search_path` for nix search paths (`option_definition.py`)

fixes https://github.com/nix-gui/nix-gui/issues/203

Managed to get the `(modulesPath + "/installer/scan/not-detected.nix")` import working, and Nix-Gui shows `hardware.enableRedistributableFirmware`'s setting from said import. (`lib.mkDefault true` isn't shows in a user friendly way right now though).

image

```
INFO:Retrieving option values for module "/etc/nixos/configuration.nix"
INFO:Retrieving option values for module "/etc/nixos/hardware-configuration.nix"
INFO:Retrieving option values for module "/etc/nixos/nixpkgs/nixos/modules/installer/scan/not-detected.nix"
INFO:Retrieving option values for module "/etc/nixos/system-specific.nix"
...
```

### CODE REVIEW

**Comment by lapp0** [comment]
@milahu please review

File: nixui/nix/lib.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
 let
   inherit (import <nixpkgs> {}) pkgs lib;
   inherit (import <nixpkgs/nixos> {}) config;
+  modulesPath = <nixpkgs/nixos/modules>;
```
Comment (milahu) [inline nixui/nix/lib.nix:4]: i guess only `pkgs` should be taken from `<nixpkgs>` and the others from `<nixos>`
`nixos` and `nixpkgs` can be separate channels, for example, stable nixos + unstable nixpkgs

modulesPath is defined in [nixpkgs/nixos/lib/eval-config.nix](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/eval-config.nix#L63)

evalModules is defined in [nixpkgs/lib/modules.nix](https://github.com/NixOS/nixpkgs/blob/master/lib/modules.nix#L86)

Comment (lapp0) [reply nixui/nix/lib.nix:4]: So it makes more sense for it to be the following?

```
  inherit (import <nixpkgs> {}) pkgs;
  inherit (import <nixpkgs/nixos> {}) config lib;
  modulesPath = <nixpkgs/nixos/modules>;
```

Comment (milahu) [reply nixui/nix/lib.nix:4]:
```nix
let
  inherit (import <nixpkgs> {}) pkgs lib;
  inherit (import <nixos/nixos> {}) config;
  modulesPath = <nixos/nixos/modules>;
in
```

or


```nix
let
  inherit (import <nixpkgs> {}) pkgs lib;
  inherit (import <nixos> {}) config;
  modulesPath = <nixos/modules>;
in
```

Comment (lapp0) [reply nixui/nix/lib.nix]: fixed by https://github.com/nix-gui/nix-gui/pull/210/commits/8c3089f4f133b647afe11e7de79d06de8659d9fd

File: nixui/nix/lib.nix [hunk 2]
```diff
@@ -1,6 +1,6 @@
 let
   inherit (import <nixpkgs> {}) pkgs lib;
-  inherit (import <nixpkgs/nixos> {}) config;
+  inherit (import <nixos> {}) config;
   modulesPath = <nixpkgs/nixos/modules>;
```
Comment (milahu) [inline nixui/nix/lib.nix:4]:
```
modulesPath = <nixos/modules>;
```

Comment (lapp0) [reply nixui/nix/lib.nix]: resolved by https://github.com/nix-gui/nix-gui/pull/210/commits/3f32e25750fd855457afe68f8e9b2228229c66a1

Comment (milahu) [reply nixui/nix/lib.nix]: lets hope this works ; )

Comment (lapp0) [reply nixui/nix/lib.nix]: All tests pass and smoke test looks fine.

**Pull request merged**